src/openms/include/OpenMS/ANALYSIS/ID/IDMapper.h

```diff
@@ -166,17 +172,13 @@ namespace OpenMS
         while (identifications_iterator != identifications_precursors.end() &&
                (identifications_iterator->first - experiment_iterator->first) < rt_tolerance_) // fabs() not required here, since are definitely within left border, and wait until exceeding the right
         {
-          // testing whether the m/z fits
-          if (!map[experiment_iterator->second].getPrecursors().empty() || mapMS1)
+          if (mapMS1 || 
+               // testing whether the m/z fits
+               ((!map[experiment_iterator->second].getPrecursors().empty()) &&
+                fabs(peptide_ids[identifications_iterator->second].getMZ() - map[experiment_iterator->second].getPrecursors()[0].getMZ()) < mz_tolerance_))
```

Comment:
Looks like absolute tolerance is compared? I thought IDMapper supports ppm? Is this a bug?